awx/api/serializers.py

```diff
@@ -4760,97 +4764,23 @@ def check_organization_permission(self, attrs, request):
         # - If the orgs is not set, set it to the org of the launching user
         # - If the user is part of multiple orgs, throw a validation error saying user is part of multiple orgs, please provide one
         if not request.user.is_superuser:
+            read_org_qs = Organization.accessible_pk_qs(request.user, 'read_role')
             if 'organization' not in attrs or attrs['organization'] == None or attrs['organization'] == '':
-                if Organization.accessible_pk_qs(request.user, 'read_role').count() == 1:
-                    for tup in Organization.accessible_pk_qs(request.user, 'read_role').all():
-                        attrs['organization'] = Organization.objects.filter(id__in=str(tup[0])).first()
-                elif Organization.accessible_pk_qs(request.user, 'read_role').count() > 1:
+                read_org_ct = read_org_qs.count()
+                if read_org_ct == 1:
+                    attrs['organization'] = read_org_qs.first()
+                elif read_org_ct > 1:
                     raise serializers.ValidationError("User has permission to multiple Organizations, please set one of them in the request")
                 else:
                     raise serializers.ValidationError("User not part of any organization, please assign an organization to assign to the bulk job")
             else:
-                allowed_orgs = set()
+                allowed_orgs = set(read_org_qs.values_list('id', flat=True))
```

Comment:
this is not giving the list of org id's but its giving list of role id's 
```
In [15]: Organization.accessible_pk_qs(User(1), 'read_role').values_list('id', flat=True)
Out[15]: <QuerySet [50, 225]>
```

Reply:
may be we need this
```
In [36]: set(Organization.accessible_pk_qs(User(1), 'read_role').values_list('object_id', flat=True))
Out[36]: {1, 2}
```

Reply:
To follow up - `accessible_pk_qs` doesn't give a flat list in python, but the `__in` syntax will work in nested expressions in queries. I know, this is weird, but it's correct.

While this could be used as a id values list query, that was never my intent. Even if you did that, you would just wind up running it again to get the objects in another context. Also, due to the extremely non-obvious gotchas I just mentioned, we should avoid having serializers touch `pk_qs` anything.

So I changed in correspondence with my intent, which is to use `accessible_objects`.